worker/cache.go

```diff
@@ -0,0 +1,231 @@
+package worker
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	"go.sia.tech/renterd/api"
+	"go.sia.tech/renterd/webhooks"
+)
+
+const (
+	cacheKeyDownloadContracts = "downloadcontracts"
+	cacheKeyGougingParams     = "gougingparams"
+
+	cacheEntryExpiry = 5 * time.Minute
```

Comment:
Ideally if we are subscribed and the subscription works, we should be good with a longer expiry. e.g. 30 minutes or even more. Because we just can't miss an update.

Of course that implies everything is working. Which brings me to an idea about a f/u which involves comparing the fetched data returned after an entry expires and making sure it matches. Because if it doesn't match it means we weren't properly updated by the bus.

There are probably a number of good reasons why the updates might not come in (e.g. network config) and we should become aware of those.

Reply:
👍  ok I'll be sure to F/U with that or a similar mechanism that checks whether we can trust the cache or not